src/ripple/app/tx/impl/Taker.cpp

```diff
@@ -169,6 +189,27 @@ qual_mul (STAmount const& amount, Quality const& quality, STAmount const& output
     return std::min (result, output);
 }
 
+void
+BasicTaker::log_flow (std::string description, Flow const& flow)
```

Comment:
Consider a template on the string type. You often pass in `char*` and we can pass this to the stream without converting to a string first.

If you don't use a template, change to `std::string const&`